services/autoscaling/src/simcore_service_autoscaling/utils_docker.py

```diff
@@ -2,163 +2,219 @@
+    )
+
+
+@retry(
+    stop=stop_after_delay(_TIMEOUT_WAITING_FOR_NODES_S),
+    before_sleep=before_sleep_log(logger, logging.WARNING),
+    after=after_log(logger, logging.ERROR),
+    wait=wait_fixed(5),
+)
+async def wait_for_node(node_name: str) -> Node:
     async with aiodocker.Docker() as docker:
+        list_of_nodes = await docker.nodes.list(filters={"name": node_name})
+    if not list_of_nodes:
+        raise TryAgain
+    return parse_obj_as(Node, list_of_nodes[0])
 
-        services = await docker.services.list()
-        for service in services:
-            tasks = service.tasks()
-            for task in tasks:
-                if (
-                    task["Status"]["State"] == "running"
-                    and task["NodeID"] in nodes_ids
-                    and (
-                        reservations := task["Spec"]
-                        .get("Resources", {})
-                        .get("Reservations")
-                    )
-                ):
-                    tasks_resources.append(
-                        {"ID": task["ID"], **_get_reserved_resources(reservations)}
-                    )
-
-                elif (
-                    task["Status"]["State"] == "pending"
-                    and task["Status"]["Message"] == "pending task scheduling"
-                    and "insufficient resources on" in task["Status"]["Err"]
-                    and (
-                        reservations := task["Spec"]
-                        .get("Resources", {})
-                        .get("Reservations")
-                    )
-                ):
-                    tasks_pending_resources.append(
-                        {"ID": task["ID"], **_get_reserved_resources(reservations)}
-                    )
-
-        total_tasks_cpus = 0
-        total_tasks_ram = 0
-        for task in tasks_resources:
-            total_tasks_cpus += task["CPU"]
-            total_tasks_ram += task["RAM"]
-
-        for task in tasks_pending_resources:
-            total_pending_tasks_cpus += task["CPU"]
-            total_pending_tasks_ram += task["RAM"]
-
-        return TasksResources(
-            total_cpus_running_tasks=total_tasks_cpus,
-            total_ram_running_tasks=total_tasks_ram,
-            total_cpus_pending_tasks=total_pending_tasks_cpus,
-            total_ram_pending_tasks=total_pending_tasks_ram,
-            count_tasks_pending=len(tasks_pending_resources),
+
+async def tag_node(node: Node, *, tags: dict[str, str], available: bool) -> None:
+    async with aiodocker.Docker() as docker:
+        assert node.ID  # nosec
+        assert node.Version  # nosec
+        assert node.Version.Index  # nosec
+        assert node.Spec  # nosec
+        assert node.Spec.Role  # nosec
+        await docker.nodes.update(
+            node_id=node.ID,
+            version=node.Version.Index,
+            spec={
+                "Availability": "active" if available else "drain",
+                "Labels": tags,
```

Comment:
I don't really get the point here. Are you telling me you will be messing around with the autoscaled nodes? in terms of labels? then we should talk as this would be a new information for me.
Otherwise, the nodes that get updated here are **only** the newly added nodes not any of the non-scaled ones.